### ISSUE TITLE
[ally]: PasswordInput - HCM - Show/Hide button icon not visible on hover

### ISSUE THREAD

**Issue by mariyageorge01** [carbon-design-system/carbon#19183]
When you enable Windows High Contrast Mode, the Show/Hide password button icon in `PasswordInput` is not visible.

https://github.com/user-attachments/assets/487b7782-535b-49e0-8a94-24abb5669fb8

**Comment by mariyageorge01** [comment]
Raised a PR: https://github.com/carbon-design-system/carbon/pull/19291 for this issue.
